src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,245 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+//
+// This class is a port of the Mono managed implementation of the MD4 algorithm
+// and required to support NTLM in Android only.
+// It's an implementation detail and is not intended to be a public API.
+// Assuming that NTLM would be System.Net.Security, it makes sense to put MD4 here as well.
+//
+namespace System.Net.Security
+{
+    internal sealed class MD4
+    {
+        internal static void HashData(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            Debug.Assert(destination.Length == 128 >> 3);
+            // Span<byte> buffer = stackalloc byte[64];
+            Span<uint> state = stackalloc uint[4];
+
+            // the initialize our context
+            count[0] = 0;
+            count[1] = 0;
+            state[0] = 0x67452301;
+            state[1] = 0xefcdab89;
+            state[2] = 0x98badcfe;
+            state[3] = 0x10325476;
+            // Zeroize sensitive information
+            Array.Clear(buffer, 0, 64);
+            Array.Clear(x, 0, 16);
+
+            HashCore(source.ToArray(), 0, source.Length, state);
+            HashValue = HashFinal(state);
+            byte[] result = (byte[]) HashValue.Clone();
+            result.CopyTo(destination);
```

Comment:
HashFinal should be changed to write to destination directly, eliminating the array creation, and the redundant clone.